tachyon/zk/plonk/examples/example_circuit_tests.cc

```diff
@@ -0,0 +1,527 @@
+  // clang-format on
+};
+template <typename TestingType>
+class ExampleCircuitTest
+    : public CircuitTest<typename TestingType::PCS, typename TestingType::LS> {
+ public:
+  static void SetUpTestSuite() { math::bn254::BN254Curve::Init(); }
+};
+
+using TestingTypes = testing::Types<
+    TestingArguments<SimpleCircuit<SHPlonk::Field, SimpleFloorPlanner>, SHPlonk,
+                     Halo2LS>,
+    TestingArguments<SimpleCircuit<SHPlonk::Field, V1FloorPlanner>, SHPlonk,
+                     Halo2LS>>;
+}  // namespace
+
+TYPED_TEST_SUITE(ExampleCircuitTest, TestingTypes);
+
+TYPED_TEST(ExampleCircuitTest, Configure) {
+  using TestingType = TypeParam;
+  using Circuit = typename TestingType::Circuit;
+  using PCS = typename TestingType::PCS;
+  using LS = typename TestingType::LS;
+  using F = typename PCS::Field;
+  using ExampleTestData =
+      typename ExampleTestDataSelector<Circuit, PCS, LS>::type;
+
+  ConstraintSystem<F> constraint_system;
+  auto config = Circuit::Configure(constraint_system);
+
+  ExampleTestData::TestConfig(config);
+
+  halo2::PinnedConstraintSystem<F> pinned_constraint_system(constraint_system);
+  EXPECT_EQ(ExampleTestData::kPinnedConstraintSystem,
+            base::ToRustDebugString(pinned_constraint_system));
+
+  EXPECT_TRUE(constraint_system.selector_map().empty());
+  EXPECT_TRUE(constraint_system.general_column_annotations().empty());
+}
+
+TYPED_TEST(ExampleCircuitTest, Synthesize) {
+  using TestingType = TypeParam;
+  using Circuit = typename TestingType::Circuit;
+  using PCS = typename TestingType::PCS;
+  using LS = typename TestingType::LS;
+  using F = typename PCS::Field;
+  using Domain = typename PCS::Domain;
+  using Commitment = typename PCS::Commitment;
+  using RationalEvals = typename PCS::RationalEvals;
+  using ExampleTestData =
+      typename ExampleTestDataSelector<Circuit, PCS, LS>::type;
+
+  CHECK(this->prover_->pcs().UnsafeSetup(ExampleTestData::kN, F(2)));
+  this->prover_->set_domain(Domain::Create(ExampleTestData::kN));
+  const Domain* domain = this->prover_->domain();
+
+  ConstraintSystem<F> constraint_system;
+  auto config = Circuit::Configure(constraint_system);
+  Assembly<RationalEvals> assembly =
+      VerifyingKey<F, Commitment>::template CreateAssembly<RationalEvals>(
+          domain, constraint_system);
+
+  Circuit circuit = ExampleTestData::GetCircuit();
+  typename Circuit::FloorPlanner floor_planner;
+  floor_planner.Synthesize(&assembly, circuit, std::move(config),
+                           constraint_system.constants());
+
+  {
+    std::vector<RationalEvals> expected_fixed_columns =
+        this->CreateRationalColumns(
+            base::Array2DToVector2D(ExampleTestData::kFixedColumnsOther));
+    EXPECT_EQ(assembly.fixed_columns(), expected_fixed_columns);
+  }
+  {
+    std::vector<AnyColumnKey> expected_columns =
+        base::ArrayToVector(ExampleTestData::kColumns);
+    EXPECT_EQ(assembly.permutation().columns(), expected_columns);
+  }
```

Comment:
Why did you put them into additional scope?

Reply:
Though not particularly necessary, I wanted to drop the variables `expected_...` right after they were being used.